### ISSUE TITLE
mod_pubsub unable to process register-push-apns command for mod_push: undefined

### ISSUE THREAD

**Issue by saribe0** [processone/ejabberd#2024]
> What version of ejabberd are you using?

ejabberd 17.08.32

> What operating system (version) are you using?

Ubuntu 16.04.3 LTS

> How did you install ejabberd (source, package, distribution)?

If I remember correctly, using Ubuntu's package manager.

> What did not work as expected? Are there error messages in the log? What
> was the unexpected behavior? What was the expected result?

Offline push notifications via apns, mod_push, and mod_pubsub is not working. Oddly, I believe the error is before mod_push or apns even gets involved and has to do with mod_pubsub not recognizing the adhoc_command to register a user. My configuration and the error is below. I can't seem to find anything about the error online so any thoughts or suggestions would be greatly appreciated.

Configuration in yaml file:
![screen shot 2017-09-30 at 11 24 22 pm](https://user-images.githubusercontent.com/6515393/31051451-b1f7320e-a636-11e7-9b27-dca2c2b31e28.png)

Error message from ejabberd log file:
![screen shot 2017-09-30 at 11 02 15 pm](https://user-images.githubusercontent.com/6515393/31051419-8e407a10-a635-11e7-8494-623c555c4ac4.png)

**Comment by zinid** [comment]
According to your configuration you're using third-party `mod_push` module, it will not work in 17.08. Recent ejabberd versions are shipped with standard `mod_push` module, which can be configured as simple as:
```
mod_push: {}
```

**Comment by saribe0** [comment]
In that case, how would I go about configuring the module to use my certificate for apns notifications and to register my users with device tokens? I can't seem to fine anything on how to get the standard mod_push module working so I was assuming it was the same as the stand alone module from previous versions.

**Comment by zinid** [comment]
Ha, you need to write your own Push APNS service, like did developers of Conversations or Chatsecure for example.
`ejabberd` doesn't provide APNS service frontend.

**Comment by saribe0** [comment]
Ok.. so what does the standard mod_push module do at the moment? How can I tap into it to write my own apns service?

**Comment by zinid** [comment]
Your service needs to support the stuff mod_push publishes, as described in [XEP-0357, Section 7](https://xmpp.org/extensions/xep-0357.html#publishing)

**Comment by saribe0** [comment]
I've been looking into XEP-0357 as you mentioned and will put together some type of service to handle the output of mod_push. My goal right now is to get mod_push to publish this output. 

When I try to enable push notifications per the specification (XEP-0357, Section 5), I receive "No module is handling this query".

I've reset mod_push and mod_pubsub in my configuration file back to the defaults:
```
mod_pubsub:
    access_createnode: pubsub_createnode
    ## reduces resource comsumption, but XEP incompliant
    ignore_pep_from_offline: true
    ## XEP compliant, but increases resource comsumption
    ## ignore_pep_from_offline: false
    last_item_cache: false
    plugins:
      - "flat"
      - "hometree"
      - "pep" # pep requires mod_caps
mod_push: {}
```

Is there something else I have to do to enable to module and have it handle the enable request?

**Comment by weiss** [comment]
> My goal right now is to get mod_push to publish this output.

Sounds unrelated to your original problem.  If you're seeing this with the `mod_push` shipped with ejabberd 17.09, please open a new GitHub issue and include the exact query you're sending.

**Comment by saribe0** [comment]
Created a new [issue here](https://github.com/processone/ejabberd/issues/2033) for mod_push shipped with ejabberd and the "No module is handling this query" error while enabling push notifications.
